### ISSUE TITLE
About SeedsNet train

### ISSUE THREAD

**Issue by zyx3216825** [BubblyYi/Coronary-Artery-Tracking-via-3D-CNN-Classification#15]
Hi ,      
I used cta08 image00-image06 as train set and leave the last image07 as validata set,  and used seedpoints_patch_generater_postive.py(gap_size=100) to generate about 80000+ positive patches, used seedpoints_patch_generater_negative.py(move_step=19) to generate about 3000+ negative patches.
The train loss (positive and nagetiave patches from image00-image06) could reduce normally, but val loss (positive and negative patches at image07) couldn't reduce , like #8 . I noticed author replied about potisive patches and negative patches balance, so I reset move_step = 10 to generate more negative patches, but val loss also couldn't reduce.  I don't know the reason, could you offer the seednet checkpoint which I can check my patches

**Comment by BubblyYi** [comment]
Please try to draw the data distribution map of blood vessel radius. You should ensure that the data distribution of training set and test set is consistent. Other indicators are the same, including offset samples and so on.

**Comment by qianjinfighter** [comment]
Hello,
Could you share CAT08 Dataset for research?  The website link of this dataset is invalid now.
My email is qianjinmingliang@163.com,  or could you share the Cloud Drive  please

**Comment by XuyangPan** [comment]
> Hello, Could you share CAT08 Dataset for research? The website link of this dataset is invalid now. My email is [qianjinmingliang@163.com](mailto:qianjinmingliang@163.com), or could you share the Cloud Drive please

你好，数据集搞定了吗，如果可以的话，能否发送相关数据或链接到我的邮箱（[xuyangpan@yeah.net](mailto:xuyangpan@yeah.net)），谢谢！

**Comment by tiantianaiqi** [comment]
您好，请问您找到数据了么，可以分享给我一份么，邮箱1519045476@qq.com，谢谢您！

**Comment by tiantianaiqi** [comment]
> > Hello, Could you share CAT08 Dataset for research? The website link of this dataset is invalid now. My email is [qianjinmingliang@163.com](mailto:qianjinmingliang@163.com), or could you share the Cloud Drive please
> 
> 你好，数据集搞定了吗，如果可以的话，能否发送相关数据或链接到我的邮箱（[xuyangpan@yeah.net](mailto:xuyangpan@yeah.net)），谢谢！

您好，请问您找到数据了么，可以分享给我一份么，邮箱1519045476@qq.com，谢谢您！

**Comment by tiantianaiqi** [comment]
> Hello, Could you share CAT08 Dataset for research? The website link of this dataset is invalid now. My email is [qianjinmingliang@163.com](mailto:qianjinmingliang@163.com), or could you share the Cloud Drive please

您好，请问您找到数据了么，可以分享给我一份么，邮箱1519045476@qq.com，谢谢您！

**Comment by tiantianaiqi** [comment]
Hello, please find the data, can you share a copy with me, the mailbox 1519045476@qq.com, thank you!

**Comment by tiantianaiqi** [comment]
Hi, did you find the dataset? , can you share it with me? My email is 1519045476@qq.com! Thank you!

**Comment by tiantianaiqi** [comment]
> > Hello, Could you share CAT08 Dataset for research? The website link of this dataset is invalid now. My email is [qianjinmingliang@163.com](mailto:qianjinmingliang@163.com), or could you share the Cloud Drive please
> 
> 你好，数据集搞定了吗，如果可以的话，能否发送相关数据或链接到我的邮箱（[xuyangpan@yeah.net](mailto:xuyangpan@yeah.net)），谢谢！

Hi, did you find the dataset? , can you share it with me? My email is 1519045476@qq.com! Thank you!

**Comment by tiantianaiqi** [comment]
> Hello, Could you share CAT08 Dataset for research? The website link of this dataset is invalid now. My email is [qianjinmingliang@163.com](mailto:qianjinmingliang@163.com), or could you share the Cloud Drive please

Hi, did you find the dataset? , can you share it with me? My email is 1519045476@qq.com! Thank you!

**Comment by tiantianaiqi** [comment]
> Hi , I used cta08 image00-image06 as train set and leave the last image07 as validata set, and used seedpoints_patch_generater_postive.py(gap_size=100) to generate about 80000+ positive patches, used seedpoints_patch_generater_negative.py(move_step=19) to generate about 3000+ negative patches. The train loss (positive and nagetiave patches from image00-image06) could reduce normally, but val loss (positive and negative patches at image07) couldn't reduce , like #8 . I noticed author replied about potisive patches and negative patches balance, so I reset move_step = 10 to generate more negative patches, but val loss also couldn't reduce. I don't know the reason, could you offer the seednet checkpoint which I can check my patches

Hi, did you find the dataset? , can you share it with me? My email is 1519045476@qq.com! Thank you!

**Comment by simi-chen** [comment]
> Hi , I used cta08 image00-image06 as train set and leave the last image07 as validata set, and used seedpoints_patch_generater_postive.py(gap_size=100) to generate about 80000+ positive patches, used seedpoints_patch_generater_negative.py(move_step=19) to generate about 3000+ negative patches. The train loss (positive and nagetiave patches from image00-image06) could reduce normally, but val loss (positive and negative patches at image07) couldn't reduce , like #8 . I noticed author replied about potisive patches and negative patches balance, so I reset move_step = 10 to generate more negative patches, but val loss also couldn't reduce. I don't know the reason, could you offer the seednet checkpoint which I can check my patches

I found that converting a central line regression network into a classification network significantly improves the performance. For positive samples, data is randomly cropped within a 4mm offset from the central line, and labeled as 1. For negative samples, data is cropped more than 4mm away from the central line, and labeled as 0. Each set of data contains about 3000 positive samples and 3000 negative samples. Using a combination of Dice and cross-entropy loss, the network training converges.
